src/BlitSort.jl

```diff
@@ -0,0 +1,470 @@
+            else
+                blit_partition!(array, ptc + 1, swap, swap_index, swap_size, quad4, cmp)
+            end
+        end
+
+        if cmp(array[pta], array[pta+1]) ≤ 0
+            if cmp(array[ptc], array[ptc+1]) ≤ 0
+                cmp(array[ptb], array[ptb+1]) ≤ 0 && return
+            else
+                rotate_merge_block!(array, array_index + asInt(half1), swap, swap_index, swap_size, quad3, quad4, cmp)
+            end
+        else
+            rotate_merge_block!(array, array_index, swap, swap_index, swap_size, quad1, quad2, cmp)
+            cmp(array[ptc], array[ptc+1]) > 0 &&
+                rotate_merge_block!(array, array_index + asInt(half1), swap, swap_index, swap_size, quad3, quad4, cmp)
+        end
+        rotate_merge_block!(array, array_index, swap, swap_index, swap_size, half1, half2, cmp)
+    end
+end
+
+# The next 4 functions are used for pivot selection
+
+function blit_binary_median(arraya, pta::Int, arrayb, ptb::Int, len::UInt, cmp::F) where {F}
+    @inbounds begin
+        while !iszero(len ÷= 2)
+            leni = asInt(len)
+            if cmp(arraya[pta+leni], arrayb[ptb+leni]) ≤ 0
+                pta += leni
+            else
+                ptb += leni
+            end
+        end
+        aa = arraya[pta]
+        bb = arrayb[ptb]
+        return ifelse(cmp(aa, bb) > 0, aa, bb)
+    end
+end
+
+function blit_trim_four!(array, pta::Int, cmp::F) where {F}
+    @inbounds begin
+        x = cmp(array[pta], array[pta+1]) > 0
+        array[pta], array[pta+1] = array[pta+x], array[pta+!x]
+        pta += 2
+
+        x = cmp(array[pta], array[pta+1]) > 0
+        array[pta], array[pta+1] = array[pta+x], array[pta+!x]
+        pta -= 2
+
+        x = 2(cmp(array[pta], array[pta+2]) ≤ 0)
+        array[pta+2] = array[pta+x]
+        pta += 1
+        x = 2(cmp(array[pta], array[pta+2]) > 0)
+        array[pta] = array[pta+x]
+    end
+end
+
+function blit_median_of_nine!(array, array_index::Int, swap, swap_index::Int, nmemb::UInt, cmp::F) where {F}
+    @inbounds begin
+        z = asInt(nmemb ÷ 9)
+
+        pta = array_index
+
+        for x in swap_index:swap_index+8
+            swap[x] = array[pta]
+            pta += z
+        end
+
+        blit_trim_four!(swap, swap_index, cmp)
+        blit_trim_four!(swap, swap_index + 4, cmp)
+
+        swap[swap_index]   = swap[swap_index+5]
+        swap[swap_index+3] = swap[swap_index+8]
+
+        blit_trim_four!(swap, swap_index, cmp)
+
+        swap[swap_index] = swap[swap_index+6]
+
+        x = cmp(swap[swap_index], swap[swap_index+1]) > 0
+        y = cmp(swap[swap_index], swap[swap_index+2]) > 0
+        z = cmp(swap[swap_index+1], swap[swap_index+2]) > 0
+
+        return swap[swap_index + (x == y) + (y ⊻ z)]
+    end
+end
+
+Base.@assume_effects :nothrow function blit_median_of_cbrt!(array, array_index::Int, swap, swap_index::Int, swap_size::UInt,
+    nmemb::UInt, cmp::F) where {F}
```

Comment:
This effect annotation is invalid in the case of a `cmp` function that may throw.